backend/main.py

```
@@ -42,396 +43,327 @@
         
-        # Construct stack path
-        stack_path = f"{compose_path.rstrip('/')}/{stack_name}"
+        # Get pools using zfs list command
+        try:
+            validated_cmd = SecurityUtils.validate_zfs_command_args("list", "-H", "-o", "name", "-t", "pool")
+            returncode, stdout, stderr = await zfs_service.run_command(validated_cmd)
+            
+            pools = []
+            if returncode == 0:
+                pools = [line.strip() for line in stdout.strip().split('\n') if line.strip()]
+        except:
```

Flake8 (7.2.0)

[error] do not use bare 'except'

(E722)